### ISSUE TITLE
ERROR NOK: ("int() argument must be a string, a bytes-like object or a number, not 'NoneType'",)

### ISSUE THREAD

**Issue by joebeem** [fail2ban/fail2ban#2771]
Hello, I have been pulling my hair out all day today trying to fix an issue I've come across and I'm not sure if it's something I'm doing or some other issue. 

I upgraded to Ubuntu 20.04 recently and since then I havent been able to get fail2ban to work correctly even though it worked for years on other versions of the operating system. 

fail2ban will start just fine, but it won't recognize a specific jail I was using for port scanning using Fail2Ban and UFW. I have pasted everything I could think of that would be needed below. 

Would anyone be able to tell me if I'm  doing something wrong or maybe something changed in the new version of OS which might be causing this?

**jail: /etc/fail2ban/jail.d/ufw-port-scan.conf** 
```
[ufw-port-scan]

enabled = true
port = all
filter = ufw-port-scan
banaction = ufw-action
logpath = /var/log/ufw.log
findtime  = 24h
bantime  = 48h
maxretry = 4
ignoreip = 10.50.0.0/16 67.47.0.0/16 24.53.0.0/16
```
**filter: /etc/fail2ban/filter.d/ufw-port-scan.conf**
```
[Definition]

failregex = .*\[UFW BLOCK\] IN=.* SRC=<HOST>
ignoreregex =
```
**action: /etc/fail2ban/action.d/ufw-action.conf**
```
[Definition]
actionstart =
actionstop =
actioncheck =
actionban = ufw insert 1 deny from  to any
actionunban = ufw delete deny from  to any
```

I thought maybe it was the regex, but upon testing it it seems to work.

```
    fail2ban-regex  /var/log/ufw.log '.*\[UFW BLOCK\] IN=.* SRC=<HOST>'

    Running tests
    =============
    
    Use   failregex line : .*\[UFW BLOCK\] IN=.* SRC=<HOST>
    Use         log file : /var/log/ufw.log
    Use         encoding : UTF-8
    
    
    Results
    =======
    
    Failregex: 2967 total
    |-  #) [# of hits] regular expression
    |   1) [2967] .*\[UFW BLOCK\] IN=.* SRC=<HOST>
    `-
    
    Ignoreregex: 0 total
    
    Date template hits:
    |- [# of hits] date format
    |  [2970] {^LN-BEG}(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
    `-
```
I then ran the following command to try to isolate where exactly the issue is taking place and it seems as though it is of course the jail I'm trying to use.
```
    sudo fail2ban-client -vvv -x stop
    sudo fail2ban-client -vvv -x start

    ...................... I cut out a lot of stuff here, but can add it if needed

    +  163 7F88A4336740 fail2ban                  HEAVY CMD: ['server-stream', [['set', 'syslogsocket', 'auto'], ['set', 'loglevel', 'INFO'], ['set', 'logtarget', '/var/log/fail2ban.log'], ['set', 'dbfile', '/var/lib/fail2ban/fail2ban.sqlite3'], ['set', 'dbmaxmatches', 10], ['set', 'dbpurgeage', '1d'], ['add', 'sshd', 'auto'], ['set', 'sshd', 'usedns', 'warn'], ['set', 'sshd', 'prefregex', None], ['set', 'sshd', 'maxlines', None], ['set', 'sshd', 'datepattern', None], ['set', 'sshd', 'maxretry', 4], ['set', 'sshd', 'findtime', '242'], ['set', 'sshd', 'bantime', '605000'], ['set', 'sshd', 'ignorecommand', ''], ['set', 'sshd', 'addignoreip',  '10.50.0.0/16', '67.47.0.0/16', '24.53.0.0/16'], ['set', 'sshd', 'logencoding', 'auto'], ['set', 'sshd', 'addlogpath', '/var/log/auth.log', 'head'], ['set', 'sshd', 'addaction', 'iptables-multiport'], ['multi-set', 'sshd', 'action', 'iptables-multiport', [['actionstart', '<iptables> -N f2b-sshd\n<iptables> -A f2b-sshd -j RETURN\n<iptables> -I INPUT -p tcp -m multiport --dports ssh -j f2b-sshd'], ['actionstop', '<iptables> -D INPUT -p tcp -m multiport --dports ssh -j f2b-sshd\n<iptables> -F f2b-sshd\n<iptables> -X f2b-sshd'], ['actionflush', '<iptables> -F f2b-sshd'], ['actioncheck', "<iptables> -n -L INPUT | grep -q 'f2b-sshd[ \\t]'"], ['actionban', '<iptables> -I f2b-sshd 1 -s <ip> -j <blocktype>'], ['actionunban', '<iptables> -D f2b-sshd -s <ip> -j <blocktype>'], ['name', 'sshd'], ['bantime', '605000'], ['port', 'ssh'], ['protocol', 'tcp'], ['chain', 'INPUT'], ['actname', 'iptables-multiport'], ['blocktype', 'REJECT --reject-with icmp-port-unreachable'], ['returntype', 'RETURN'], ['lockingopt', '-w'], ['iptables', 'iptables <lockingopt>'], ['blocktype?family=inet6', 'REJECT --reject-with icmp6-port-unreachable'], ['iptables?family=inet6', 'ip6tables <lockingopt>']]], ['add', 'ufw-port-scan', 'auto'], ['set', 'ufw-port-scan', 'usedns', 'warn'], ['set', 'ufw-port-scan', 'addfailregex', "'.*\\[UFW BLOCK\\] IN=.* SRC=<HOST>'"], ['set', 'ufw-port-scan', 'maxretry', 4], ['set', 'ufw-port-scan', 'findtime', '24h'], ['set', 'ufw-port-scan', 'bantime', '48h'], ['set', 'ufw-port-scan', 'ignorecommand', ''], ['set', 'ufw-port-scan', 'addignoreip', '10.50.0.0/16', '67.47.0.0/16', '24.53.0.0/16'], ['set', 'ufw-port-scan', 'logencoding', 'auto'], ['set', 'ufw-port-scan', 'addlogpath', '/var/log/ufw.log', 'head'], ['set', 'ufw-port-scan', 'addaction', 'ufw-action'], ['multi-set', 'ufw-port-scan', 'action', 'ufw-action', [['actionstart', ''], ['actionstop', ''], ['actioncheck', ''], ['actionban', 'ufw insert 1 deny from  to any'], ['actionunban', 'ufw delete deny from  to any'], ['name', 'ufw-port-scan'], ['bantime', '168h'], ['port', 'all'], ['protocol', 'tcp'], ['chain', 'INPUT'], ['actname', 'ufw-action']]], ['start', 'sshd'], ['start', 'ufw-port-scan']]]
    +  176 7F88A4336740 fail2ban                  ERROR NOK: ("int() argument must be a string, a bytes-like object or a number, not 'NoneType'",)

    +  176 7F88A4336740 fail2ban                  HEAVY CMD: ['server-status']
    +  177 7F88A4336740 fail2ban                  HEAVY OK : 'Server ready'
    +  177 7F88A4336740 fail2ban.beautifier       HEAVY Beautify 'Server ready' with ['server-status']
    Server ready
```

Thank you for taking the time to read and possibly respond to this message. Apologize if I broke any rules that needed to be followed.

**Comment by sebres** [comment]
Please provide your fail2ban version.

In the stream excerpt, I see some parameters that are indeed `None`, for instance:
```
..., ['set', 'sshd', 'maxlines', None], ['set', 'sshd', 'datepattern', None] ,...
```
This leads to assumption that your `/etc/fail2ban` is wrong (misconfigured somewhere), e. g. `jail.conf` may be empty (or completely overwritten by you) etc pp (`maxlines` is normally set in sshd-filter, also it will be never None in normal case if supplied in stream).
1. If you had changed stock `.conf` files - please don't do that - if required, any customizations must be done in `.local` files.
Make a backup of `/etc/fail2ban` and restore original config from fail2ban distribution or set some defaults like in original `jail.conf`.
2. If `.conf` files are unmodified - check whether your fail2ban-client version matches with installed (e. g. may be it is PATH conflict, and you have still some old version that will be found as first in path).
Also check `whereis fail2ban-client`, `fail2ban-python --version` and `fail2ban-python -c 'from fail2ban.version import version; print(version)'`.

Additionally check whether you can find some errors in fail2ban.log? I cannot say what exactly caused this as long as I don't have your configuration.
Anyway I don't think your jail or filter `ufw-port-scan` are reason for that, since I see no errors there at all.

**Comment by joebeem** [comment]
Thank you for replying. I am very happy to say that I was able to resolve the issue with your advice. 

I was aware of some of the things you mentioned, but most especially not changing the .conf files (use local files). Why first tried troubleshooting the issue, one of the first things I tried was completely removing the act all together including any configuration files. I would then install a fresh version directly from my repository currently set up on the server. I'm not really sure where the issue was, but after creating a backup, I again removed everything using apt remove fail2ban --purge and removed /etc/fail2ban. However, this time I decided to use the git method outlined on the homepage to install  the application.  After it completed the setup process, I copied my configuration files back to the directory and suddenly it worked! 

@sebres Thank you very much for your assistance!
